src/main/java/hudson/plugins/git/UserRemoteConfig.java

```diff
@@ -127,7 +129,7 @@ public FormValidation doCheckCredentialsId(@AncestorInPath Item project,
             return FormValidation.ok();
         }
 
-        public FormValidation doCheckUrl(@AncestorInPath Item project,
+        public FormValidation doCheckUrl(@AncestorInPath Job project,
```

Comment:
UI validation is what end-user will see, and I don't like the idea he get false negative validation while the build is ok.